context/app/static/js/components/entity-search/searchkit-modifications/useSearchkitSDK.js

```diff
@@ -1,35 +1,48 @@
 import { useState } from 'react';
 import Searchkit from '@searchkit/sdk';
 import useDeepCompareEffect from 'use-deep-compare-effect';
+import RequestTransporter from 'js/components/entity-search/searchkit-modifications/RequestTransporter';
 
 // Copied from https://github.com/searchkit/searchkit/blob/6d11b204520009a705fe207535bd4f18d083d361/packages/searchkit-sdk/src/react-hooks/index.ts
 // Modified to handle initial filters and use our custom transformer
 
-const useSearchkitSDK = ({ config, variables, transporter, filters, defaultSort }) => {
+const useSearchkitSDK = ({ config, variables, filters, defaultSort }) => {
   const [results, setResponse] = useState(null);
   const [loading, setLoading] = useState(true);
 
   useDeepCompareEffect(() => {
+    const abortController = new AbortController();
+    const transporter = new RequestTransporter(config, abortController);
+
     async function fetchData(v) {
       setLoading(true);
       const request = Searchkit(config, transporter)
         .query(v.query)
         .setFilters([...filters.map((filter) => filter.value), ...v.filters])
         .setSortBy(v.sortBy || defaultSort);
 
-      const response = await request.execute({
-        facets: true,
-        hits: {
-          size: variables.page.size,
-          from: variables.page.from,
-        },
-      });
-      setLoading(false);
-      setResponse(response);
+      try {
+        const response = await request.execute({
+          facets: true,
+          hits: {
+            size: variables.page.size,
+            from: variables.page.from,
+          },
+        });
+        setLoading(false);
+        setResponse(response);
+      } catch (error) {
+        if (error.name === 'AbortError') {
+          // Aborting throws an error and will prevent state from being updated.
```

Comment:
```suggestion
        if (error.name !== 'AbortError') {
          // Aborting throws an error and will prevent state from being updated.
          // Other errors should still be thrown
          throw error;
```
I'm not sure about the bigger picture, but right now it's a no-op.

Reply:
It's a required no-op. You need a `catch` with `try`.